app/interactors/tasks/destroy.rb

```diff
@@ -0,0 +1,19 @@
+module Tasks
+  class Destroy
+    include Interactor
+
+    delegate :task, to: :context
+
+    def call
+      destroy_task_or_fail
+    end
+
+    private
+
+    def destroy_task_or_fail
+      unless task.destroy
+        context.fail!(error: "Failed to destroy task")
```

Comment:
Все круто сделано, но в Руби можно и так:
```suggestion
      context.fail!(error: "Failed to destroy task") unless task.destroy
```
Более красиво и лакончино